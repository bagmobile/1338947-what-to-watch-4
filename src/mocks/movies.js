export const mockMovies = [
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Crime`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 54,
    isFavorite: false,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Crime`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 87,
    isFavorite: true,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
  {
    id: 3,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Comedies`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 154,
    isFavorite: true,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
  {
    id: 4,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    genre: `Dramas`,
    year: 2011,
    ratingScore: 4.4,
    ratingCount: 200,
    description: `description`,
    director: `Director`,
    starring: [`Star 1`, `Star 2`, `Star 3`, `Star 4`, `Star 5`, `Star 6`, `Star 7`, `Star 8`],
    duration: 67,
    isFavorite: true,
    backgroundColor: `#A6B7AC`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"`,
  },
];

export const genres = [`All genres`, `Dramas`, `Comedies`, `Romance`, `Sci-Fi`, `Crime`, `Documentary`];
