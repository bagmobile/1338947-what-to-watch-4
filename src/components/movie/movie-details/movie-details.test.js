import React from "react";
import renderer from "react-test-renderer";
import {mockMovies} from "../../../mocks/movies.js";
import MovieDetails from "./movie-details";

const movie = mockMovies[0];

describe(`MovieDetails component`, () => {

  it(`Render`, () => {
    const tree = renderer
      .create(<MovieDetails
        movie={movie}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
