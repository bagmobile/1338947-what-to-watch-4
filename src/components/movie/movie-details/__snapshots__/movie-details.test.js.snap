// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component Render 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Director
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        <span>
          Star 1
          <br />
        </span>
        <span>
          Star 2
          <br />
        </span>
        <span>
          Star 3
          <br />
        </span>
        <span>
          Star 4
          <br />
        </span>
        <span>
          Star 5
          <br />
        </span>
        <span>
          Star 6
          <br />
        </span>
        <span>
          Star 7
          <br />
        </span>
        <span>
          Star 8
          <br />
        </span>
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        54m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Crime
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2011
      </span>
    </p>
  </div>
</div>
`;
